### PR TITLE
undo RFC #294, it breaks code for no good reason, the compiler can wa…

### PR DESCRIPTION
**Pull request by Araq**
…rn about the construct instead

### CODE REVIEW

**Comment by konsumlamm** [comment]
Link to the RFC: https://github.com/nim-lang/RFCs/issues/294.

File: changelog.md [hunk 1]
```diff
@@ -23,15 +23,14 @@
 - `repr` now doesn't insert trailing newline; previous behavior was very inconsistent,
   see #16034. Use `-d:nimLegacyReprWithNewline` for previous behavior.
 
-- An enum now can't be converted to another enum directly, you must use `ord` (or `cast`, but
-  compiler won't help if you misuse it).
+- A type conversion from one enum type to another now produces a `[EnumConv]` warning.
+  You should use `ord` (or `cast`, but compiler won't help if you misuse it).
```
Comment (konsumlamm) [inline changelog.md]:
```suggestion
- A type conversion from one enum type to another now produces an `[EnumConv]` warning.
  You should use `ord` (or `cast`, but the compiler won't help, if you misuse it) instead.
```

File: tests/misc/tconv.nim [hunk 2]
```diff
@@ -1,7 +1,4 @@
 discard """
-  nimout:'''
-tconv.nim(81, 15) Warning: enum to enum conversion is now deprecated [User]
-'''
```
Comment (konsumlamm) [inline tests/misc/tconv.nim:4]: Why not keep a test for the warning?

File: tests/misc/tconv.nim [hunk 3]
```diff
@@ -64,25 +61,3 @@ block: # issue 3766
       echo x
     r 3.R
 
-
```
Comment (timotheecour) [inline tests/misc/tconv.nim]: at least apply this patch instead (it works, and ensures the warning works as designed)

```
diff --git a/tests/misc/tconv.nim b/tests/misc/tconv.nim
index ebd718f71..38a47f646 100644
--- a/tests/misc/tconv.nim
+++ b/tests/misc/tconv.nim
@@ -1,4 +1,5 @@
 discard """
+  matrix: "--warningAsError:EnumConv"
 """

 template reject(x) =
@@ -61,3 +62,24 @@ block: # issue 3766
       echo x
     r 3.R

+block: # https://github.com/nim-lang/RFCs/issues/294
+  type Koo = enum k1, k2
+  type Goo = enum g1, g2
+
+  accept: Koo(k2)
+  accept: k2.Koo
+  accept: k2.int.Goo
+
+  reject: Goo(k2)
+  reject: k2.Goo
+  reject: k2.string
+
+  {.push warningAsError[EnumConv]:off.}
+  discard Goo(k2)
+  accept: Goo(k2)
+  accept: k2.Goo
+  reject: k2.string
+  {.pop.}
+
+  reject: Goo(k2)
+  reject: k2.Goo
```

File: tests/misc/tconv.nim (hunk 2 above)
Comment (timotheecour) [reply tests/misc/tconv.nim:4]: see https://github.com/nim-lang/Nim/pull/18546#discussion_r673501303 which does exactly that

**Comment by timotheecour** [comment]
ok, a warning is a good way to handle this.

**Pull request merged**